File: software/bindings/c/example_simple.c
// tested with 
// gcc -lpthread -lrt -o example_simple bricklet_dual_relay.c 
//     ip_connection.c example_simple.c
// on Ubuntu

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_dual_relay.h"

#define HOST "localhost"
#define PORT 4223
#define UID "xyz" // Change to your UID

int main() {
	// Create ip connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	DualRelay dr;
	dual_relay_create(&dr, UID); 

	// Add device to ip connection
	if(ipcon_add_device(&ipcon, &dr) < 0) {
		fprintf(stderr, "Could not connect to Brick\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Turn relay 1 on and relay 2 off.
	dual_relay_set_state(&dr, true, false);

	printf("Press ctrl+c to close\n");
	ipcon_join_thread(&ipcon); // Join mainloop of ip connection
}

File: software/bindings/java/ExampleSimple.java
import com.tinkerforge.BrickletDualRelay;
import com.tinkerforge.IPConnection;

public class ExampleSimple {
	private static final String host = new String("localhost");
	private static final int port = 4223;
	private static final String UID = new String("ABC"); // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(host, port); // Create connection to brickd (Can throw IOException)

		BrickletDualRelay dr = new BrickletDualRelay(UID); // Create device object
		ipcon.addDevice(dr); // Add device to ip connection (Can throw IPConnection.TimeoutException)
		// Don't use device before it is added to a connection
		

		// Turn relays alternating on/off for 10 times with 1 second delay
		for(int i = 0; i < 10; i++) {
			Thread.sleep(1000);
			if(i % 2 == 1) {
				dr.setState(true, false);
			} else {
				dr.setState(false, true);
			}
		}

		System.out.println("Press ctrl+c to exit");
		ipcon.joinThread();
	}
}
